src/plugins/ui/evaluation_chart.ts

```diff
@@ -61,20 +66,31 @@ export class EvaluationChartPlugin extends UIPlugin {
    *
    *  - the chart background color if one is defined
    *  - the fill color of the cell if one is defined
+   *  - the text color of the cell if one is defined
    *  - the fill color of the cell from conditional formats if one is defined
+   *  - the text color of the cell from conditional formats if one is defined
    *  - the default chart color if no other color is defined
+   *  - the default text color if no other color is defined
    */
-  getBackgroundOfSingleCellChart(
+  getStyleOfSingleCellChart(
     chartBackground: Color | undefined,
     mainRange: Range | undefined
-  ): Color {
-    if (chartBackground) return chartBackground;
+  ): EvaluationChartStyle {
+    if (chartBackground)
+      return { background: chartBackground, fontColor: chartFontColor(chartBackground) };
     if (!mainRange) {
-      return BACKGROUND_CHART_COLOR;
+      return {
+        background: BACKGROUND_CHART_COLOR,
+        fontColor: chartFontColor(BACKGROUND_CHART_COLOR),
+      };
     }
     const col = mainRange.zone.left;
     const row = mainRange.zone.top;
-    const style = this.getters.getCellComputedStyle(mainRange.sheetId, col, row);
-    return style.fillColor || BACKGROUND_CHART_COLOR;
+    const sheetId = mainRange.sheetId;
+    const style = this.getters.getCellComputedStyle(sheetId, col, row);
+    return {
+      background: style.fillColor || BACKGROUND_CHART_COLOR,
+      fontColor: style.textColor || chartFontColor(style.fillColor),
```

Comment:
shouldn't it fallback on BACKGROUND_CHART_COLOR
```suggestion
      fontColor: style.textColor || chartFontColor(style.fillColor || BACKGROUND_CHART_COLOR),
```
`style.fillColor || BACKGROUND_CHART_COLOR` can be extracted to a local `background` variable.

Reply:
Yep makes sense, I went with the approach of sending "null" value to it, so that it can auto assign the default color. But this one looks more understandable. :+1: